logstash-core/src/main/java/org/logstash/plugins/PluginFactoryExt.java

```diff
@@ -395,16 +397,16 @@ public ExecutionContextExt create(final ThreadContext context, final IRubyObject
         }
 
         public Context toContext(PluginLookup.PluginType pluginType, AbstractNamespacedMetricExt metric) {
-            DeadLetterQueueWriter dlq = null;
+            DeadLetterQueueWriter dlq = NullDeadLetterQueueWriter.getInstance();
             if (pluginType == PluginLookup.PluginType.OUTPUT) {
                 if (dlqWriter instanceof AbstractDeadLetterQueueWriterExt.PluginDeadLetterQueueWriterExt) {
                     IRubyObject innerWriter =
                             ((AbstractDeadLetterQueueWriterExt.PluginDeadLetterQueueWriterExt) dlqWriter)
                                     .innerWriter(RubyUtil.RUBY.getCurrentContext());
 
                     if (innerWriter != null) {
-                        if (innerWriter.getJavaClass().equals(DeadLetterQueueWriter.class)) {
-                            dlq = innerWriter.toJava(DeadLetterQueueWriter.class);
+                         if (DeadLetterQueueWriter.class.isAssignableFrom(innerWriter.getJavaClass())) {
```

Comment:
One of these is fully qualified (`org.logstash.common.io.DeadLetterQueueWriter`) and the other isn't (`DeadLetterQueueWriter`), which means it'll resolve to the imported `co.elastic.logstash.api.DeadLetterQueueWriter`; are we casting to the same thing we're checking?

Reply:
Good catch!